Tests/InProcessInteroperabilityTests/InProcessInteroperabilityTests.swift

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+import GRPCInProcessTransport
+import XCTest
+
+@testable import InteroperabilityTests
+
+final class InProcessInteroperabilityTests: XCTestCase {
+  func runInProcessTransport(
+    interopTestCase: InteroperabilityTestCase
+  ) async throws {
+    do {
+      let inProcess = InProcessTransport.makePair()
+      try await withThrowingTaskGroup(of: Void.self) { group in
+        group.addTask {
+          let server = GRPCServer(transports: [inProcess.server], services: [TestService()])
+          try await server.run()
+        }
+
+        group.addTask {
+          try await withThrowingTaskGroup(of: Void.self) { clientGroup in
+            let client = GRPCClient(transport: inProcess.client)
+            clientGroup.addTask {
+              try await client.run()
+            }
+            try await interopTestCase.makeTest().run(client: client)
+
+            clientGroup.cancelAll()
+          }
+        }
+
+        try await group.next()
+        group.cancelAll()
+      }
+    } catch let error as AssertionFailure {
+      XCTFail(error.message)
+    }
+  }
+
+  func testEmtyUnary() async throws {
+    try await self.runInProcessTransport(interopTestCase: InteroperabilityTestCase.emptyUnary)
```

Comment:
nit: prefer using type inference here 
```suggestion
    try await self.runInProcessTransport(interopTestCase: .emptyUnary)
```